File: 643 - Maximum Average Subarray I.py
class Solution:
    def findMaxAverage(self, nums, k):
        end = len(nums) - k
        s = sum(nums[0:k])
        res = s / k
        print(s, res)
        for i in range(1, end + 1):
            s = s + nums[i + k - 1] - nums[i - 1]
            res = max(s / k, res)
            print(s, res)
        return res


s = Solution()
s.findMaxAverage([1, 12, -5, -6, 50, 3], 4)

File: 643 - Maximum Average Subarray I.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        double sum = 0;
        int end = nums.length - k;
        for (int i = 0; i < k; i++) {
            sum += nums[i];
        }
        double res = sum / k;
        for (int i = 1; i <= end; i++) {
            sum = sum - nums[i - 1] + nums[i + k - 1];
            res = Math.max(res, sum / k);
        }
        return res;
    }
}